### DISCUSSION TITLE
파일 최대 크기 검증

### DISCUSSION THREAD

**Discussion by Kim0914** [Busan-Dinosaur/Foodbowl-Backend#112] [의논해볼까요?]
저희 서버에서는 최대 파일 크기를 제한하고 있습니다.

컨트롤러 테스트를 MockMvc를 이용해 진행하고 있는데, mockMvc는 스프링 파일 최대 크기 설정 환경과는 분리된 환경에서 동작하기 때문에 테스트하기 어렵다고 합니다 !

해당 검증을 추가할 수 있으면 추가하는게 좋다고 생각하는데, 어떻게 풀어갈 수 있을지 의논해보면 좋을 것 같아요 !

**Comment by woo-chang** [comment]
꼼꼼한 테스트 너무 좋습니다 👍

우선 테스트 `application.yml` 파일에 `spring.servlet.multipart.max-file-size`, `spring.servlet.multipart.max-request-size`를 설정했음에도 해당 설정이 적용되지 않는지 궁금합니다 🤔

만약 된다면 테스트에는 설정값을 작게 가져가 설정값을 초과하는 `Mock file`을 만들어서 테스트해볼 수 있지 않을까?라는 생각이 들어요.

설정값을 초과하는 요청이 들어왔을 때 `MaxUploadSizeExceededException`이 발생하는데, `GlobalExceptionHandler`에서 해당 예외를 잡아 예외 응답을 출력하도록 하고 해당 응답이 잘 보여지는지 확인하는 테스트를 작성하면 되지 않을까라고 생각하고 있습니다 ㅎㅎ

안된다면 조금 더 찾아보고 고민이 필요한 부분인 것 같네요~

**Reply by Kim0914** [reply]
`application.yml`에 max-file-size, max-request-size 설정이 되어있는 상태에서
컨트롤러 테스트에서 50MB MockMultipartFile을 만들어서 테스트 했는데 그대로 통과가 되더라구요 😂

그래서 찾아보니 mockMvc에서는 실제 서블릿 컨테이너를 사용하지 않아서 해당 설정이 적용되지 않는다고 합니다.

말씀해주신 것 처럼 파일 크기가 넘는 경우 `MaxUploadSizeExceededException`이 발생하는데 해당 예외를 터트리도록 mocking한 다음 검증하는 방향은 어떨까요?!

**Reply by woo-chang** [reply]
mockMvc는 설정이 적용되지 않는군요! 하나 배워갑니다 ㅎㅎ

> 말씀해주신 것 처럼 파일 크기가 넘는 경우 MaxUploadSizeExceededException이 발생하는데 해당 예외를 터트리도록 mocking한 다음 검증하는 방향은 어떨까요?!

실제 서블릿 컨테이너 적용없이 테스트할 수 있는 좋은 방법인 것 같습니다. 진행시켜주시죠 🔥
